confidence_checks/synthetic_data.py

```diff
@@ -0,0 +1,36 @@
+from typing import Tuple
```

Comment:
I'd suggest renaming this file to `toy_data.py` given how small it is. `synthetic_data` often implies that this is a synthetic dataset used to demonstrate results in a public-facing way (e.g. in a paper or talk).